src/diart/models.py

```diff
@@ -168,8 +206,55 @@ def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__(PyannoteLoader(model_info, hf_token))
 
     def forward(
-        self,
-        waveform: torch.Tensor,
-        weights: Optional[torch.Tensor] = None,
+            self,
+            waveform: torch.Tensor,
+            weights: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
         return self.model(waveform, weights=weights)
+
+
+class WeSpeakerSpeakerEmbeddingInference(LazyWeSpeakerSpeakerEmbedding):
+    """Minimal interface for a we speaker embedding inference."""
+
+    def __init__(self, loader: Callable[[], WeSpeakerPretrainedSpeakerEmbedding]):
+        super().__init__(loader)
+
+    @staticmethod
+    def from_pyannote(inference,
+                      ) -> "WeSpeakerSpeakerEmbeddingInference":
+        """
+        Returns an `EmbeddingModel` wrapping a pyannote model.
+
+        Parameters
+        ----------
+        inference: pyannote.audio.pipelines.speaker_verification.WeSpeakerPretrainedSpeakerEmbedding
+            The pyannote.audio inference to fetch.
+
+        Returns
+        -------
+        wrapper: EmbeddingModel
+        """
+        assert _has_pyannote, "No pyannote.audio installation found"
+        return PyannoteWeSpeakerSpeakerEmbeddingInference(inference)
+
+    def eval(self):
+        """Mock method to match pytorch api"""
+        return
+
+
+class PyannoteWeSpeakerSpeakerEmbeddingInference(WeSpeakerSpeakerEmbeddingInference):
```

Comment:
This class shouldn't exist. This logic should be implemented in `PyannoteEmbeddingModel.__call__()` (`forward()` should be replaced because `LazyModel` will not be a `nn.Module` anymore)